File: LiveDataSampleJava/app/src/main/java/com/websarva/wings/android/livedatasamplejava/data/local/AppDatabase.java
package com.websarva.wings.android.livedatasamplejava.data.local;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {Cocktailmemo.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {
	private static AppDatabase _instance;

	public static AppDatabase getDatabase(Context context) {
		if (_instance == null) {
			_instance = Room.databaseBuilder(context.getApplicationContext(), AppDatabase.class, "cocktailmemo_db").build();
		}
		return _instance;
	}

	public abstract CocktailmemoDAO createCocktailmemoDAO();
}

File: LiveDataSampleKotlin/app/src/main/java/com/websarva/wings/android/livedatasamplekotlin/data/local/AppDatabase.kt
package com.websarva.wings.android.repositorysamplekotlin.data.local

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [Cocktailmemo::class], version = 1)
abstract class AppDatabase : RoomDatabase() {
	companion object {
		private var _instance: AppDatabase? = null

		fun getDatabase(context: Context): AppDatabase {
			if (_instance == null) {
				_instance = Room.databaseBuilder(context.applicationContext, AppDatabase::class.java, "cocktailmemo_db").build()
			}
			return _instance!!
		}
	}

	abstract fun createCocktailmemoDAO(): CocktailmemoDAO
}
